keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/AuthorizationTestSupport.kt

```diff
@@ -0,0 +1,66 @@
+package com.netflix.spinnaker.keel.rest
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import dev.minutest.ContextBuilder
+import io.mockk.every
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+/**
+ *  This function loops through the specified API requests and permissions and adds a test for each,
+ *  so that you don't have to manually write those tests in your REST controller test classes.
+ */
+fun ContextBuilder<*>.testApiPermissions(
```

Comment:
I understand that your goal here is to make it so we don't have to repeat code for tests. However, I am having a really hard time (even with your comments) understanding what this is testing, how it's testing it, and how to make changes or diagnose failures if I'm refactoring. I think that it should be very obvious what tests are testing - for me, looping and auto-generating things is much harder to understand than a verbose test or a series of similar tests. I prefer the tests that you've deleted - where the api and the description or behavior is explicit.

Reply:
I appreciate your concern with readability and maintainability of the tests. However, a question I would ask back to you is: how often do you go back and refactor tests, especially those for  behavior that you take for granted once put in place, such as these authorization checks?

What I attempted to give you here is almost like a small DSL where you can actually clearly see what APIs are being tested and with which permissions, as you can see in the example below (or the one I gave you above):
https://github.com/spinnaker/keel/blob/2ac3cb709e91f0ca50c5773e8293d8e25fcefb0f/keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/ResourceControllerTests.kt#L116-L135

I frankly cannot see how this is harder to understand than a bunch of `test` blocks that specify those API calls and checks, but also have a lot of that boilerplate code I abstracted away.

Reply:
P.S. Instead of this:
```kotlin
      ApiRequest("GET /resources/${resource.id}") to setOf(
        Permission(APPLICATION_AUTHZ, READ, RESOURCE),
        Permission(CLOUD_ACCOUNT_AUTHZ, READ, RESOURCE)
      )
```
...is this easier to read?
```kotlin
      ApiRequest("GET /resources/${resource.id}") to setOf(
        ApplicationPermission(READ, RESOURCE),
        CloudAccountPermission(READ, RESOURCE)
      )
```